sql/driver/driver_it_test.go

```diff
@@ -460,6 +460,35 @@ func TestSetSSLConfig(t *testing.T) {
 	}
 }
 
+func TestContextCancelAfterFirstPage(t *testing.T) {
```

Comment:
If we test context cancellation is working with manual cancellation will it always work in other context cancellation variants such as timeout? 

If so, an easier and more reliable approach might be to cancel the context manually in the for loop after the first row. I am assuming the first page will contain just one row since server generates one record in a second. 

The reason I suggested this approach is the fact that initiating a query, sending the request to server and receiving and parsing the response from server can take more than 1100ms time in some cases and we can cancel the result before the first page is received.  I don't think time passed being less than 2000ms proves the otherwise. Time passed can be 1900ms and the context cancellation can happen at 1100ms and first page can be received at 1700ms.

Reply:
Yes, manual cancellation and timeout cancellation uses the same mechanism.

That makes sense. Updated the test at: https://github.com/hazelcast/hazelcast-go-client/pull/732/commits/67f639ffa9e7b56914698f1beee58080daea9e8f